lib/SILOptimizer/Transforms/ArrayElementValuePropagation.cpp

```diff
@@ -12,8 +12,12 @@
 #define DEBUG_TYPE "array-element-propagation"
 
 #include "llvm/ADT/SetVector.h"
+#include "swift/AST/NameLookup.h"
+#include "swift/AST/ParameterList.h"
+#include "swift/AST/GenericEnvironment.h"
 #include "swift/SIL/SILBasicBlock.h"
 #include "swift/SIL/SILInstruction.h"
+#include "swift/SIL/SILFunction.h"
 #include "swift/SIL/DebugUtils.h"
 #include "swift/SILOptimizer/Analysis/ArraySemantic.h"
 #include "swift/SILOptimizer/PassManager/Passes.h"
```

Comment:
You should update the general comment for this optimization (some lines below) and describe the new feature.